dkist/wcs/models.py

```diff
@@ -267,6 +276,279 @@ def evaluate(self, lon, lat, z, crpix, cdelt, lon_pole, **kwargs):
+
+    @property
+    def inverse(self):
+        ivct = InverseVaryingCelestialTransform2D(crpix=self.crpix,
+                                                  cdelt=self.cdelt,
+                                                  lon_pole=self.lon_pole,
+                                                  pc_table=self.pc_table,
+                                                  crval_table=self.crval_table,
+                                                  projection=self.projection)
+        return ivct
+
+
+class InverseVaryingCelestialTransform2D(BaseVaryingCelestialTransform2D):
+    n_inputs = 4
+    n_outputs = 2
+
+    @property
+    def input_units(self):
+        return {"lon": u.deg, "lat": u.deg, "z": u.pix, "q": u.pix}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.inputs = ("lon", "lat", "z", "q")
+        self.outputs = ("x", "y")
+
+    def evaluate(self, lon, lat, z, q, crpix, cdelt, lon_pole, **kwargs):
+        return self._map_transform(lon, lat, z, q, crpix, cdelt, lon_pole,
+                                   inverse=True)
+
+
+class BaseVaryingCelestialTransformSlit(BaseVaryingCelestialTransform):
+    def _map_transform(self, x, y, crpix, cdelt, lon_pole, inverse=False):
+        # We need to broadcast the arrays together so they are all the same shape
+        bx, by = np.broadcast_arrays(x, y, subok=True)
+        # Convert the z coordinate into an index to the lookup tables
+        yind = self.sanitize_index(by)
+
+        # Generate output arrays (ignore units for simplicity)
+        if isinstance(bx, u.Quantity):
+            x_out = np.empty_like(bx.value)
+            y_out = np.empty_like(by.value)
+        else:
+            x_out = np.empty_like(bx)
+            y_out = np.empty_like(by)
+
+        # We now loop over every unique value of z and compute the transform.
+        # This means we make the minimum number of calls possible to the transform.
+        for yy in np.unique(yind):
+            # Scalar parameters are reshaped to be length one arrays by modeling
+            sct = self.transform_at_index(yy, crpix[0], cdelt[0], lon_pole[0])
+
+            # Call this transform for all values of x, y where y == yind
+            mask = yind == yy
+            if inverse:
+                # Note this isn't use as the inverse of this model uses the
+                # standard 2D inverse.
+                xx, yy = sct.inverse(bx[mask], by[mask])  # pragma: no cover
+            else:
+                xx, yy = sct(bx[mask], by[mask])
+
+            if isinstance(xx, u.Quantity):
+                x_out[mask], y_out[mask] = xx.value, yy.value
+            else:
+                x_out[mask], y_out[mask] = xx, yy
+
+        # Put the units back
+        if isinstance(xx, u.Quantity):
+            x_out = x_out << xx.unit
+            y_out = y_out << yy.unit
+
+        return x_out, y_out
+
+
+class VaryingCelestialTransformSlit(BaseVaryingCelestialTransformSlit):
```

Comment:
After thinking about the comment on line 461 below this model name confuses me. Looking at the inputs it seems like "slity" is a detector dimension and "raster" is an index (see line 375 above). That sounds like 2 spatial dimensions to me and I don't see a time dimension despite "Varying" being in the name.